tests/compiler_test.go

```diff
@@ -0,0 +1,17 @@
+// +build js
```

Comment:
We use `t.Skip` rather than build constraints for skipping tests meant for `GOARCH=js` only, unless the build constraint is necessary for things to compile.

In this case, this test shouldn't be skipped on non-`js`; I want to see it run and pass using `gc` as well (just like all other tests in this package that which can pass with `gc`).

So, remove this line.

Reply:
Done.